modules/tinymce/src/core/test/ts/browser/textpatterns/ReplacementTest.ts

```diff
@@ -150,4 +152,38 @@ describe('browser.tinymce.core.textpatterns.ReplacementTest', () => {
     // This function throws an error here due to the bug identified in TINY-8779
     TinyContentActions.keystroke(editor, Keys.enter());
   });
+
+  context('TINY-8779: Matches text nodes in a paragraph', () => {
+    let editor: Editor;
+
+    const hook = TinyHooks.bddSetupLight<Editor>({
+      text_patterns: [
+        { start: 'one_error', replacement: 'no_error' }
+      ],
+      indent: false,
+      base_url: '/project/tinymce/js/tinymce'
+    }, [ ]);
+
+    const setCursorAndPressEnterAndAssert = (editor: Editor, texts: string[], expected: string) => {
+      const targetParagraph = editor.dom.select('p')[0];
+      Arr.each(texts, (t) => targetParagraph.appendChild(document.createTextNode(t)));
+      editor.focus();
+      TinySelections.setCursor(editor, [ 0, 2 ], 6);
+      TinyContentActions.keystroke(editor, Keys.enter());
+      assert.equal(editor.getContent(), expected);
+    };
+
+    beforeEach(() => {
+      editor = hook.editor();
+      editor.setContent('<p></p>');
```

Comment:
Is there a reason this is in the `beforeEach` instead of the "test" function you have above? As it's doing so much already it might as well do this 🤷

Reply:
I must have left it there after tidying up the code. Have moved it into a better place